src/query/storage/types.go

```diff
@@ -245,6 +245,7 @@ type WriteQueryOptions struct {
 	Datapoints ts.Datapoints
 	Unit       xtime.Unit
 	Annotation []byte
+	Type       ts.PromMetricType
```

Comment:
Why is this needed, isn't the type already present in the annotation when this struct is created? If there are some cases when it isn't, I think we should fix them instead. Having two competing fields for the same purpose will only cause trouble.

Reply:
Also, where does this field get assigned?

Reply:
We assign this field when we receive data from the aggregator. Aggregator doesn't store any information about annotations: https://github.com/m3db/m3/blob/gg/aggregator-keep-metric-type/src/cmd/services/m3coordinator/ingest/m3msg/ingest.go#L222